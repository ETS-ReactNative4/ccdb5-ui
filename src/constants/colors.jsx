/* eslint-disable no-inline-comments, camelcase */
module.exports = Object.freeze( {
  BriteCharts: {
    regular: '#20aa3f',
    medium: '#ADDC91',
    light: '#C7E5B3'
  },
  DataLens: {
    0: '#2cb34a',
    1: '#addc91',
    2: '#257675',
    3: '#9ec4c3',
    4: '#0072ce',
    5: '#96c4ed',
    6: '#254b87',
    7: '#9daecc',
    8: '#b4267a',
    9: '#dc9cbf',
    10: '#a2a3a4'
  }
} )
