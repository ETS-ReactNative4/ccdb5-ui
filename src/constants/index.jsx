// Internal triggers
export const PAGE_CHANGED              = 'PAGE_CHANGED'
export const SEARCH_CHANGED            = 'SEARCH_CHANGED'
export const SIZE_CHANGED              = 'SIZE_CHANGED'
export const SORT_CHANGED              = 'SORT_CHANGED'
export const URL_CHANGED               = 'URL_CHANGED'
export const FILTER_CHANGED            = 'FILTER_CHANGED'
export const FILTER_REMOVED            = 'FILTER_REMOVED'
export const FILTER_ALL_REMOVED        = 'FILTER_ALL_REMOVED'

// External Triggers
export const COMPLAINTS_RECEIVED       = 'COMPLAINTS_RECEIVED'
export const COMPLAINTS_FAILED         = 'COMPLAINTS_FAILED'

// Useful constants
export const knownFilters = [
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tag',
  'timely',
  'zip_code'
]

export const SLUG_SEPARATOR            = 'â€¢'
