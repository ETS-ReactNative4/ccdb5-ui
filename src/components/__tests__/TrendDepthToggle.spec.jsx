import configureMockStore from 'redux-mock-store'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import ReduxTrendDepthToggle, {
  TrendDepthToggle,
  mapDispatchToProps,
  mapStateToProps
} from '../Trends/TrendDepthToggle'
import React from 'react'
import renderer from 'react-test-renderer'
import { REQUERY_ALWAYS } from '../../constants'
import { shallow } from 'enzyme'
import thunk from 'redux-thunk'

function setupEnzyme( { cbIncrease, cbReset, diff } ) {
  return shallow( <TrendDepthToggle diff={ diff }
                                    increaseDepth={ cbIncrease }
                                    lens={ 'Product' }
                                    resetDepth={ cbReset }
                                    showToggle={ true }/> )
}

function setupSnapshot( { focus, lens, productAggs } ) {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {
    aggs: {
      product: productAggs
    },
    query: {
      focus,
      lens
    },
    trends: {
      results: {
        product: [ { name: 'a', visible: true }, { name: 'b', visible: true },
          { name: 'c', visible: true }, { name: 'd', visible: true },
          { name: 'e', visible: true }, { name: 'f', visible: true },
          { name: 'g', visible: true }, { name: 'h', visible: true }
        ]
      }
    }
  } )

  return renderer.create(
    <Provider store={ store }>
      <IntlProvider locale="en">
        <ReduxTrendDepthToggle/>
      </IntlProvider>
    </Provider>
  )
}

describe( 'component:TrendDepthToggle', () => {
  let params

  beforeEach( () => {
    params = {
      focus: '',
      lens: '',
      productAggs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]
    }
  } )

  it( 'does not render when Focus', () => {
    params.focus = 'A focus item'
    const target = setupSnapshot( params )
    const tree = target.toJSON()
    expect( tree ).toBeNull()
  } )

  it( 'does not render lens is not Product', () => {
    params.lens = 'Cannot See'
    const target = setupSnapshot( params )
    const tree = target.toJSON()
    expect( tree ).toBeNull()
  } )


  it( 'renders Product view more without crashing', () => {
    params.lens = 'Product'
    const target = setupSnapshot( params )
    const tree = target.toJSON()
    expect( tree ).toMatchSnapshot()
  } )

  it( 'renders Product view fewer without crashing', () => {
    params.lens = 'Product'
    params.productAggs = [ 1, 2, 3, 4, 5 ]
    const target = setupSnapshot( params )
    const tree = target.toJSON()
    expect( tree ).toMatchSnapshot()
  } )

  describe( 'buttons', () => {
    let cbIncrease = null
    let cbReset = null
    let target

    beforeEach( () => {
      cbIncrease = jest.fn()
      cbReset = jest.fn()

    } )

    it( 'increaseDepth is called when the increase button is clicked', () => {
      target = setupEnzyme( { cbIncrease, cbReset, diff: 1000 } )
      const prev = target.find( '#trend-depth-button' )
      prev.simulate( 'click' )
      expect( cbIncrease ).toHaveBeenCalledWith( 1000 )
    } )

    it( 'reset depth is called when the reset button is clicked', () => {
      target = setupEnzyme( { cbIncrease, cbReset, diff: 0 } )
      const prev = target.find( '#trend-depth-button' )
      prev.simulate( 'click' )
      expect( cbReset ).toHaveBeenCalled()
    } )
  } )

  describe( 'mapDispatchToProps', () => {
    it( 'hooks into changeDepth', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch ).increaseDepth( 13 )
      expect( dispatch.mock.calls ).toEqual( [
        [ {
          requery: REQUERY_ALWAYS,
          depth: '18',
          type: 'DEPTH_CHANGED'
        } ]
      ] )
    } )

    it( 'hooks into resetDepth', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch ).resetDepth()
      expect( dispatch.mock.calls ).toEqual( [
        [ {
          requery: REQUERY_ALWAYS,
          type: 'DEPTH_RESET'
        } ]
      ] )
    } )

  } )

  describe( 'mapStateToProps', () => {
    it( 'maps state and props', () => {
      const state = {
        aggs: {
          product: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]
        },
        query: {
          focus: '',
          lens: 'Product'
        },
        trends: {
          results: {
            product: [ { name: 'a', visible: true }, {
              name: 'b',
              visible: true
            },
              { name: 'c', visible: true }, { name: 'd', visible: true },
              { name: 'e', visible: true }, { name: 'f', visible: true },
              { name: 'g', visible: true }, { name: 'h', visible: true }
            ]
          }
        }
      }
      let actual = mapStateToProps( state )
      expect( actual ).toEqual( {
        diff: 3,
        showToggle: true
      } )
    } )
  } )

} )
