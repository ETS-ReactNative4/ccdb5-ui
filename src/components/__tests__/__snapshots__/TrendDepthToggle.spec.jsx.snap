// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:TrendDepthToggle renders Product view fewer without crashing 1`] = `
<div
  className="trend-depth-toggle"
>
  <button
    className="a-btn a-btn__link"
    id="trend-depth-button"
    onClick={[Function]}
  >
    <span
      className="minus"
    />
    View fewer
  </button>
</div>
`;

exports[`component:TrendDepthToggle renders Product view more without crashing 1`] = `
<div
  className="trend-depth-toggle"
>
  <button
    className="a-btn a-btn__link"
    id="trend-depth-button"
    onClick={[Function]}
  >
    <span
      className="plus"
    />
    View more
  </button>
</div>
`;
